src/components/OptionList/components/Option/Option.scss

```
@@ -91,3 +91,15 @@ $control-vertical-adjustment: rem(2px);
   @include recolor-icon(var(--p-icon));
   padding: 0 spacing(tight);
 }
+
+.VerticalAlign--Top {
+  align-items: flex-start;
+}
+
+.VerticalAlign--Center {
+  align-items: center;
+}
+
+.VerticalAlign--Bottom {
+  align-items: flex-end;
+}
```

I looked at other instances where a variation pattern is used and this repo doesn't use the `{ClassName}--{Variation}` but rather `{ClassName}{Variation}`.

I suggest to change it for consistency.

---

(I personally do prefer the `--` separation though)

---

There's magic under the hood that autogenerates BEM classes in the final CSS bundled from SCSS. For prop/JS controlled variations, we use their camelcase names (e.g., for prop `verticalAlign` we'd make the class name `verticalAlign{Center/Top/Bottom}` and the output classname in the browser in dev mode would be something like `Polaris__Option--verticalAlignCenter`), you can checkout [the Stack stylesheet](https://github.com/Shopify/polaris-react/blob/b95fe67e5d33ddd271ce9729415750616bc1858d/src/components/Stack/Stack.scss#L28) for more examples of that.

```suggestion
.verticalAlignTop {
  align-items: flex-start;
}

.verticalAlignCenter {
  align-items: center;
}

.verticalAlignBottom {
  align-items: flex-end;
}
```

---

@chloerice Ah, amazing. Thanks for the clarification on this 😄